Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp

```diff
@@ -1088,6 +1088,8 @@ namespace AzToolsFramework
 
         m_listModel->SearchStringChanged(filterString);
         m_proxyModel->UpdateFilter();
+
+        m_gui->m_objectTree->expandAll();
```

Comment:
We should get UX feedback on this. @rainbj 

We shouldn't expand the entire tree, which this change would do. We could expand the minimum set so that the filtered items would be visible. We would need to also add logic though to back out the expansion if the search is cleared.

Reply:
@cgalvan I guess this applies to this bug too (same fix but in Prefab mode): https://github.com/o3de/o3de/pull/6195

Reply:
Verified this during the UI-UX meeting, and the new behavior was approved.
When searching by keyword, all entities that aren't matching are hidden. So expanding all is the correct behavior.

![searchExpand](https://user-images.githubusercontent.com/82231674/145263905-2eccf9ff-1e3a-4046-86f4-f0747c736e5f.gif)

Reply:
the EntityOutlinerListModel won't match the state of the tree if you do this.

Reply:
Good callout @pollend, what would your suggestion be for @tjmgd? Any follow-up? Thanks!

Reply:
would the expanded state be restored after the search is done? seems kind of annoying that it would be kept like that.The prefab window area uses the expand state to figure out where to draw the bottom part of the rectangle for that nested prefab window.  umm try searching for a prefab I think it will be shown cut.